### PR TITLE
Prototype for the Visualize page

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
This is still a bit rough around the edges, but it represents a realistic living example of what the visualize page can be. 

It is currently using mock data, but the mocks are modeled after the GeoJSON model described in #744 

Screen Shot 2019-09-09 at 1 18 28 AM

### CODE REVIEW

File: api/models/entity/Contribution.ts [hunk 1]
```diff
@@ -531,12 +531,43 @@ export const contributionSummaryFields = <const>[
...
 ];
 export type IContributionSummary = Pick<Contribution, typeof contributionSummaryFields[number]>;
 
+export interface IContributionGeoJson {
+    type: 'Feature';
+    properties: {
+        city: string;
+        state: string;
+        zip: string;
+        amount: number;
```
Comment (danmelton) [inline api/models/entity/Contribution.ts]: add matchAmount, oaeType

File: api/models/entity/Contribution.ts [hunk 2]
```diff
@@ -531,12 +531,43 @@ export const contributionSummaryFields = <const>[
...
+        contributorType: ContributorType;
+        contributionType: ContributionType;
+        contributionSubType: ContributionSubType;
+        date: string;
+        campaign: {
+            name: string;
+            id: string
```
Comment (danmelton) [inline api/models/entity/Contribution.ts]: add office

File: api/models/entity/Contribution.ts [hunk 3]
```diff
@@ -609,10 +643,44 @@ export async function getContributionsByGovernmentIdAsync(
...
     } catch (err) {
+        console.log(err)
         throw new Error('Error executing get contributions query');
     }
 }
 
+export function convertToGeoJson(contributions: IContributionSummaryResults): IContributionsGeoJson {
+    const data  = contributions
+        .data
+        .map((contribution: Pick<Contribution, typeof contributionSummaryFields[number]>): IContributionGeoJson => {
```
Comment (danmelton) [inline api/models/entity/Contribution.ts]: add filter for archive/draft

File: api/models/entity/Contribution.ts [hunk 4]
```diff
@@ -609,10 +643,44 @@ export async function getContributionsByGovernmentIdAsync(
...
+            return {
+                type: 'Feature',
+                properties: {
+                    city: contribution.city,
```
Comment (danmelton) [inline api/models/entity/Contribution.ts:740]: add matchAmount and oaeType

**Comment by danmelton** [comment]
This feature needs all the gis functinoality of the app, and contributions that are processed or submitted.

SO. First, let's setup the api.

You'll need a .env file (copy the /.env.example to /.env, and add the GIS key - ask dan for it)

docker-compose up api (let this run)
It'll seed the db, and populate the redis queue for our jobs service

build and run the docker jobs
docker-compose build jobs
docker-compose up jobs (let it run - it will do our geocoding)

cd app
yarn install && yarn start

navigate to frontend UI
login as govadmin@openelectionsportland.org/password
navigate to contributions, view any contribution, update the status in the upper right hand corner to submitted

you will have data at :3000/contributionsgeo

the HomePage component has an example of the redux action to grab publicData.

**Comment by jaronheard** [comment]
@danmelton I'm working on updating some of the seed data so that it has a mix of campaigns and contribution type/subtype so that we have some better sample data for the visualizations, and was wondering if setting some contribution statuses to submitted in the seed file would worry you for any reason.

I don't think it's absolutely necessary, so let me know if you think it would cause issues.

**Comment by danmelton** [comment]
> @danmelton I'm working on updating some of the seed data so that it has a mix of campaigns and contribution type/subtype so that we have some better sample data for the visualizations, and was wondering if setting some contribution statuses to submitted in the seed file would worry you for any reason.
> 
> I don't think it's absolutely necessary, so let me know if you think it would cause issues.

@jaronheard Yep, go for it.

**Comment by jaronheard** [comment]
@danmelton re-opening this since we're still working off this branch, though if you've got another suggested approach let me know.

**Pull request merged**